content/en/blog/_posts/0000-00-00-kube-proxy-non-privileged.md

```diff
@@ -0,0 +1,101 @@
+---
+layout: blog
+title: "Kubernetes 1.29: Support for Running kube-proxy in a Non-privileged Container"
+date: 0000-00-00
+slug: kube-proxy-non-privileged
+---
+
+**Author**: Lars Ekman
+
+This post describes how the `--init-only` flag to `kube-proxy`, added in K8s v1.29, can
+be used to perform configuration that requires privileged mode in an
+initContainer, while the main `kube-proxy` container may run with a
+stricter securityContext.
+
+Please note that `kube-proxy` can be installed in different ways. For
+installers such as `kubeadm` that install `kube-proxy` to run as a Pod, the new
+functionality can help you improve security.
+
+
+## Background
+
+It is undesirable to run a server container like `kube-proxy` in
+privileged mode. Security aware users wants to [use capabilities instead](
+https://github.com/kubernetes/kubernetes/issues/112171).
+
+If `kube-proxy` is installed as a POD, the initialization requires
+"privileged" mode, mostly for setting sysctl's. However, for some time
+now `kube-proxy` doesn't alter sysctl's and other settings that
+already have the desired value. In theory this can be used to do all
+privileged configuration in an [init container](/docs/concepts/workloads/pods/init-containers/).
```

Comment:
I think this would be confusing to someone who doesn't already understand the problem...

```suggestion
If `kube-proxy` is installed as a POD, the initialization requires
"privileged" mode, mostly for setting sysctl's. However, `kube-proxy`
only tries to set the sysctl's if they don't already have the right
values. In theory, then, if a privileged [init container](/docs/concepts/workloads/pods/init-containers/)
set the sysctls to the right values, then `kube-proxy` could run
unprivileged.
```